File: app/src/main/java/com/developer/kulloveth/navcomponentsample/fragments/SignUpFragment.java
package com.developer.kulloveth.navcomponentsample.fragments;


import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.developer.kulloveth.navcomponentsample.R;
import com.google.android.material.button.MaterialButton;

/**
 * A simple {@link Fragment} subclass.
 */
public class SignUpFragment extends Fragment {

    MaterialButton button;

    public SignUpFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        final View view = inflater.inflate(R.layout.fragment_sign_up, container, false);
        button = view.findViewById(R.id.sign_up_button);
        button.setOnClickListener(
                Navigation.createNavigateOnClickListener(R.id.action_signUpFragment_to_loginFragment, null));

        return view;
    }

}

File: navcomponentkotlin/src/main/java/com/developer/kulloveth/navcomponentkotlin/fragments/SignUpFragment.kt
package com.developer.kulloveth.navcomponentkotlin.fragments


import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.navigation.Navigation
import com.developer.kulloveth.navcomponentkotlin.R
import kotlinx.android.synthetic.main.fragment_sign_up.*

/**
 * A simple [Fragment] subclass.
 */
class SignUpFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
         TextView(activity).apply {
            setText(R.string.hello_blank_fragment)
        }

        return inflater.inflate(R.layout.fragment_sign_up,container,false)
    }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        signUpButton.setOnClickListener (
            Navigation.createNavigateOnClickListener(R.id.action_signUpFragment_to_loginFragment, null))

    }


}
